claat/fetch/filesystem/filesystem_test.go

```diff
@@ -0,0 +1,53 @@
+package filesystem
+
+import (
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestNewFileSystemFetcher(t *testing.T) {
+	p := "this/is/a/file/path"
+	fsf := NewFileSystemFetcher(p)
+
+	if fsf.resPath != p {
+		t.Errorf("NewFileSystemFetcher(%v).resPath = %v, want %v", p, fsf.resPath, p)
+	}
+}
+
+func TestFetch(t *testing.T) {
+	// Make temporary file for testing purposes.
+	contents := []byte("file contents!")
+	f, err := ioutil.TempFile("", "filesystem_test_file")
+	if err != nil {
+		t.Errorf("error creating temp file: %s", err)
+	}
+	fname := f.Name()
+	defer os.Remove(fname)
+
+	// Write some bytes to the file.
+	_, err = f.Write(contents)
+	if err != nil {
+		// Be sure to close the file if we're going to stop here.
+		f.Close()
```

Comment:
why not have this/`f.Close()` as a `defer` after `defer os.Remove` above if it is a wanted behavior no matter what happens? (`defer` order seems to be LIFO). then you'd get rid of the double mention?

Reply:
Good idea